R/get_ishares_index_data.R

```diff
@@ -5,8 +5,8 @@
 #' @param url A string containing the url of the desired iShares index data.
 #' @param name A string containing the name of the index.
 #' @param timestamp A string indicating the desired timestamp, by year and
-#'   quarter, "YYYYMMDD" (e.g. "20201231"). Data will be scraped for the first
-#'   day in the corresponding quarter.
+#'   quarter, "YYYYMMDD" (e.g. "20201231"). Data will be scraped for date
+#'   specified.
```

Comment:
The wording of this seems off... "by year and quarter" doesn't match the expected format of "YYYYMMDD"

Reply:
also, if we change the meaning of the argument, I would prefer we also change the name, i.e. `timestamp` was originally used to align with what we refer to as `timestamp` elsewhere in the form of "2022Q4", whereas here we want a specific date in the form "20221230", so I would suggest maybe `as_of_date` for the argument name.

Reply:
Yeah that makes sense. I was a little confused if we were actually changing the meaning of the parameter though (check out the original example function, it seems like we were always expecting `20211231`?)

I was sorta confused honestly

Reply:
https://github.com/RMI-PACTA/pacta.data.scraping/blob/3be8e19a74dd65565502482f7fd2d58bebd1a9b3/R/get_ishares_index_data.R#L20-L23

Reply:
But anyway, still totally happy to change it to `as_of_date` if that's what is preferred

Reply:
I think I had difficulties with this before (and also different use cases from different repos) and eventually I settled on a solution that can deal with either format, even though the documentation was left to suggest only one specific use case.

Reply:
Ok cool. 

I think now this function is only called in `workflow.prepare.pacta.indices` (well, at least, it is never called in `workflow.data.preparation`), so I am just going to change the parameter name, since I agree `as_of_date` seems to be more indicative of what we want.